src/init.cpp

```diff
@@ -118,7 +118,7 @@ static fs::path GetPidFile(const ArgsManager& args)
 
 [[nodiscard]] static bool CreatePidFile(const ArgsManager& args)
 {
-    fsbridge::ofstream file{GetPidFile(args)};
+    fsbridge::ofstream file{static_cast<std::filesystem::path>(GetPidFile(args))};
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/20744#discussion_r733637362

I was looking into this again, thinking passing `path.native()` might be a third alternative, but it doesn't work because there's no constructor https://en.cppreference.com/w/cpp/io/basic_ifstream/basic_ifstream that accepting the `wstring` on windows like the one accepting the wide character pointer returned by `path.c_str()`.